mongorestore/mongorestore.go

```diff
@@ -327,6 +327,7 @@ func (restore *MongoRestore) ParseAndValidateOptions() error {
 
 // Restore runs the mongorestore program.
 func (restore *MongoRestore) Restore() Result {
+	defer restore.SessionProvider.Close()
```

Comment:
What's the rationale for closing the SessionProvider when we call `Restore()`? Given that it's not opened/created in `Restore()`, this seems like surprising behavior to me

Reply:
When we run the tests, we only run `restore.Restore()` so the sessions don't get closed like they usually would if `main()` is the entry point. This was causing issues for me with testing locally since the mongod was flooded with connections and crashed. I suppose the more correct approach would be to close the connections properly in each test.